pubnub-kotlin/pubnub-kotlin-impl/src/integrationTest/kotlin/com/pubnub/api/integration/SubscribeIntegrationTests.kt

```diff
@@ -1253,6 +1253,153 @@ class SubscribeIntegrationTests : BaseIntegrationTest() {
         assertEquals(2, subscriptionSet.subscriptions.size)
     }
 
+    @Test
+    fun `shouldDeduplicateChannelSubscriptionsWhenSubscribingToSameChannelMultipleTimes`() {
+        // given
+        val numberOfSubscribe = 4
+        // punbub.subscribe does subscribe to already subscribed channel so only two subscribe calls should occur. Handshake and actual subscribe.
+        val countDownLatch = CountDownLatch(2)
+        var interceptedUrl: HttpUrl? = null
+        val testChannel = randomChannel()
+
+        clientConfig = {
+            httpLoggingInterceptor =
+                HttpLoggingInterceptor { message ->
+                    // Intercept subscribe GET request
```

Comment:
What happens if the now-deprecated `HttpLoggingInterceptor` gets removed? Are there any better alternatives like capturing outgoing HTTP requests with structured data instead of just plain text logs?